### PR TITLE
Add Building with Foundry Guide

### PR DESCRIPTION
**Pull request by briandoyle81CB**
**What changed? Why?**

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: apps/base-docs/docs/building-with-base/guides/building-with-base-and-foundry/introduction.md [hunk 1]
```diff
@@ -93,14 +93,18 @@ forge build:
 
 ## Setting up Foundry with Base
 
-In order to work with `Base`, you need to configure a couple of settings in the configuration `foundry.toml` file.
+In order to work with Base, you need to configure a couple of settings in the configuration `foundry.toml` file.
 
-The first thing is the solidity version. 
+The first thing is the solidity version.
```
Comment (wbnns) [inline apps/base-docs/docs/building-with-base/guides/building-with-base-and-foundry/introduction.md:98]: s/solidity/Solidity/

File: apps/base-docs/docs/building-with-base/guides/building-with-base-and-foundry/introduction.md [hunk 2]
```diff
@@ -137,12 +141,11 @@ We included 2 JSON RPC endpoints for `Base` and `Base goerli` and similar for th
 
 ## Conclusion
 
-In this guide, you've embarked on the journey of smart contract development with Base and Foundry. We covered the essential steps, from installing Foundry using the convenient Foundryup toolchain installer to creating your first project and configuring Foundry to seamlessly integrate with Base.
+In this guide, you've embarked on the journey of smart contract development with Base and Foundry. You've learned the essential steps, from installing Foundry using the convenient `Foundryup`` toolchain installer to creating your first project and configuring Foundry to seamlessly integrate with Base.
```
Comment (wbnns) [inline apps/base-docs/docs/building-with-base/guides/building-with-base-and-foundry/introduction.md:144]: s/Foundryup/foundryup/ 

rm extra backtick

File: apps/base-docs/docs/building-with-base/guides/building-with-base-and-foundry/introduction.md [hunk 3]
```diff
@@ -93,14 +93,18 @@ forge build:
...
-In order to work with `Base`, you need to configure a couple of settings in the configuration `foundry.toml` file.
+In order to work with Base, you need to configure a couple of settings in the configuration `foundry.toml` file.
 
-The first thing is the solidity version. 
+The first thing is the solidity version.
 
-> As of December 2023, Base does not support the new PUSH0 opcode introduced in Shanghai, which is the default target for the Solidity compiler if you use version 0.8.20 or later. See [Differences between Ethereum and Base] to learn more.
+:::caution
+
+As of December 2023, Base does not support the new PUSH0 opcode introduced in Shanghai, which is the default target for the Solidity compiler if you use version 0.8.20 or later. See [Differences between Ethereum and Base] to learn more.
```
Comment (rcbgr) [inline apps/base-docs/docs/building-with-base/guides/building-with-base-and-foundry/introduction.md:102]: Wasn't PUSH0 added in Canyon? Do we need to clarify? Is Canyon live on testnet?

**Review by rcbgr — DISMISSED** [review]

**Review by wbnns — CHANGES_REQUESTED** [review]
@briandoyle81CB 

Heya! Thanks! 

I left a few nits.

File: apps/base-docs/docs/building-with-base/guides/building-with-base-and-foundry/introduction.md (hunk 2 above)
Comment (taycaldwell) [inline apps/base-docs/docs/building-with-base/guides/building-with-base-and-foundry/introduction.md:144]: There's a duplicate backtick here after Foundryup may mess up rendering.

**Review by taycaldwell — CHANGES_REQUESTED** [review]

**Review by wbnns — APPROVED** [review]

**Review by taycaldwell — APPROVED** [review]

**Review by rcbgr — APPROVED** [review]
LGTM

**Pull request merged**